File: web/src/types/legalConsent.ts
export interface LegalConsentState {
  termsAccepted: boolean;
  healthStatementAccepted: boolean;
  privacyPolicyAccepted: boolean;
  acceptedAt: string | null;
  completed: boolean;
}

export interface SaveLegalConsentRequest {
  termsAccepted: boolean;
  healthStatementAccepted: boolean;
  privacyPolicyAccepted: boolean;
}

File: backend/src/main/java/pl/platformatreningowa/legalconsent/entity/LegalConsent.java
package pl.platformatreningowa.legalconsent.entity;

import java.time.Instant;

public record LegalConsent(
        boolean termsAccepted,
        boolean healthStatementAccepted,
        boolean privacyPolicyAccepted,
        Instant acceptedAt,
        boolean completed
) {
}
